keccak256/src/permutation/iota.rs

```diff
@@ -0,0 +1,200 @@
+use crate::arith_helpers::{convert_b2_to_b13, convert_b2_to_b9, A4};
+use crate::common::{PERMUTATION, ROUND_CONSTANTS};
+use crate::gate_helpers::biguint_to_f;
+use eth_types::Field;
+use halo2_proofs::circuit::AssignedCell;
+use halo2_proofs::circuit::Layouter;
+use halo2_proofs::{
+    plonk::{Advice, Column, ConstraintSystem, Error, Fixed, Selector},
+    poly::Rotation,
+};
+use itertools::Itertools;
+use num_bigint::BigUint;
+use std::convert::TryInto;
+
+#[derive(Clone, Debug)]
+pub struct IotaConfig<F> {
+    q_enable: Selector,
+    lane00: Column<Advice>,
+    flag: Column<Advice>,
+    round_constant: Column<Fixed>,
+    round_constant_b13: F,
+    round_constants_b9: [BigUint; PERMUTATION],
```

Comment:
Why does the config need to store the constants?

We already know them right? So we can just take them directly from the `constants` file where they're defined.

Reply:
The type of F is known at the configuration time, and we eventually need to convert them to F at some point. Config is the earliest possible place to define them.